transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java

```diff
@@ -334,7 +334,7 @@ private boolean doWriteMessage(Object msg) throws Exception {
 
         if (remoteAddress == null) {
             remoteAddress = remote;
-            if (remoteAddress == null) {
+            if (remoteAddress == null || !isConnected()) {
```

Comment:
this code was entirely removed in `EpollDatagramChannel` and instead we use `writeAddress` if `remoteAddress == null`. Please make these classes consistent or add a comment as to why they are not consistent. Also should this have been caught by a unit test?